app/models/manageiq/providers/google/container_manager.rb

```diff
@@ -0,0 +1,57 @@
+ManageIQ::Providers::Kubernetes::ContainerManager.include(ActsAsStiLeafClass)
+
+class ManageIQ::Providers::Google::ContainerManager < ManageIQ::Providers::Kubernetes::ContainerManager
+  require_nested :Container
+  require_nested :ContainerGroup
+  require_nested :ContainerNode
+  require_nested :EventCatcher
+  require_nested :EventParser
+  require_nested :Refresher
+  require_nested :RefreshWorker
+
+  def self.ems_type
+    @ems_type ||= "gke".freeze
+  end
+
+  def self.description
+    @description ||= "Google Kubernetes Engine".freeze
+  end
+
+  def self.display_name(number = 1)
+    n_('Container Provider (Google)', 'Container Providers (Google)', number)
+  end
+
+  def self.default_port
+    443
+  end
+
+  def self.kubernetes_auth_options(options)
+    {
+      :bearer_token => google_access_token(options[:bearer])
+    }
+  end
+
+  def self.google_access_token(json_key)
+    require "googleauth"
+
+    client_email, private_key, project_id, quota_project_id =
+      JSON.parse(json_key).values_at("client_email", "private_key", "project_id", "quota_project_id")
+
+    private_key = OpenSSL::PKey::RSA.new(private_key)
```

Comment:
Can the key only be rsa or could there be, say, ed25519 keys?

Reply:
This comes from the google-auth library (see https://github.com/ManageIQ/manageiq-providers-google/pull/195#discussion_r687183246)

I wanted to see if we could bypass the weirdness of having to pass in an IO when we already have the string.  The extra logic that I had to copy doesn't seem worth it though